### PR TITLE
proto file with scalar variable for sensors

### PR DESCRIPTION
**Pull request by vturpin**
Proponents: 
Moderator: @OceanGlidersCommunity/format-mantainers

# Type of PR

- [ ] Typo without possible change of interpretation of the related text.
- [ ] Fix of some error, inconsistency, unforeseen limitation.
- [ ] Style that only affects visually the compiled document
- [ ] Addition that does not require change in the current structure.
- [x] Enhancement that require changes to improve the format.

# Related Issues

# Dates when it got review approvals

# Release checklist

- [ ] Approved by at least two members of the committee?
- [ ] There were modifications after the review approvals? If so, please
      ask reviewers to update their review.
- [ ] Proponents and moderador should explicitly agree that it is ready to
      to merge.
- [ ] The moderador is the one in charge to actually merge or close this PR
      according to the final decision.

# For maintainers

- [ ] Update the moderator with a volunteer from the committee. It would be
      best to have one single moderator to guide and help this PR to move
      forward. It is OK to update the moderador pass it to another one.
- [ ] Confirm that the associated branch was deleted after the merging.
- [ ] Wrap-up and close the related issues.

# Comments

Format will be adjusted to this proto file after discussion

### CODE REVIEW

File: sp041_20191205T1757-instrument-scalar_v2.cdl [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+		PRES_QC:long_name = "quality flag" ;
+	float DEPTH(N_MEASUREMENTS) ;
+		DEPTH:_FillValue = NaN ;
+		DEPTH:long_name = "Depth" ;
+		DEPTH:vocabulary = "http://vocab.nerc.ac.uk/collection/OG1/current/DEPTH/";
+		DEPTH:standard_name = "depth" ;
+		DEPTH:units = "m" ;
+		DEPTH:ancillary_variables = "DEPTH_QC" ;
+		DEPTH:positive = "down" ;
+        DEPTH:sensor = "/CTD";
```
Comment (kerfoot) [inline sp041_20191205T1757-instrument-scalar_v2.cdl:98]: This currently points to the old CTD group. Should point to the INSTRUMENT_CTD variable, i.e.: DEPTH:sensor = "INSTRUMENT_CTD".

File: sp041_20191205T1757-instrument-scalar_v2.cdl [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+        TEMP:vocabulary = "http://vocab.nerc.ac.uk/collection/OG1/current/TEMP/";
+		TEMP:standard_name = "sea_water_temperature" ;
+		TEMP:units = "Celsius" ;
+		TEMP:ancillary_variables = "TEMP_QC" ;
+		TEMP:valid_max = 40. ;
+		TEMP:valid_min = -5. ;
+		TEMP:coverage_content_type = "physicalMeasurement" ;
+		TEMP:coordinates = "time lon lat depth" ;
+		TEMP:comment = "Sea temperature in-situ ITS-90 scale" ;
+        TEMP:sensor = "/CTD";
```
Comment (kerfoot) [inline sp041_20191205T1757-instrument-scalar_v2.cdl:114]: This currently points to the old CTD group. Should point to the INSTRUMENT_CTD variable, i.e.: DEPTH:sensor = "INSTRUMENT_CTD".

File: sp041_20191205T1757-instrument-scalar_v2.cdl [hunk 3]
```diff
@@ -0,0 +1,417 @@
...
+		CHLA:_FillValue = NaN ;
+		CHLA:long_name = "Chlorophyll-a concentration" ;
+        CHLA:vocabulary = "http://vocab.nerc.ac.uk/collection/OG1/current/CHLA/";
+		CHLA:standard_name = "mass_concentration_of_chlorophyll_a_in_sea_water" ;
+		CHLA:units = "mg m-3" ;
+		CHLA:ancillary_variables = "CHLA_QC" ;
+		CHLA:coverage_content_type = "physicalMeasurement" ;
+		CHLA:coordinates = "time lon lat depth" ;
+		CHLA:comment = "In-situ fluorometer with either manufacturer, laboratory or sample calibration applied" ;
+        CHLA:sensor = "/FLUOROMETER_CHLA";
```
Comment (kerfoot) [inline sp041_20191205T1757-instrument-scalar_v2.cdl:144]: This currently points to the old CHLA group. Should point to the INSTRUMENT_CHLA variable, i.e.: CHLA:sensor = "INSTRUMENT_CHLA".

File: sp041_20191205T1757-instrument-scalar_v2.cdl [hunk 4]
```diff
@@ -0,0 +1,417 @@
...
+		DOXY:long_name = "Dissolved oxygen" ;
+        DOXY:vocabulary = "http://vocab.nerc.ac.uk/collection/OG1/current/DOXY/";
+		DOXY:standard_name = "moles_of_oxygen_per_unit_mass_in_sea_water" ;
+		DOXY:units = "micromol kg-1" ;
+		DOXY:ancillary_variables = "DOXY_QC" ;
+		DOXY:valid_max = 0. ;
+		DOXY:valid_min = 500. ;
+		DOXY:coverage_content_type = "physicalMeasurement" ;
+		DOXY:comment = "Concentration of dissolved oxygen per unit mass of the water column. Oxygen may be expressed in terms of mass, volume or quantity of substance" ;
+        DOXY:sensor = "/OPTODE_DOXY";
```
Comment (kerfoot) [inline sp041_20191205T1757-instrument-scalar_v2.cdl:159]: This currently points to the old OPTODE_OXY group. Should point to the INSTRUMENT_OPTODE_DOXY variable, i.e.: DOXY:sensor = "INSTRUMENT_OPTODE_DOXY".

**Review by kerfoot — CHANGES_REQUESTED** [review]
There are GEOPHYSICAL_PARAMETER:sensor attributes that point to the old group format (i.e.: GEOPHYSICAL_PARAMETER:sensor = "/GROUP"). They need to be changed to point to the scalar instrument variables (i.e.: GEOPHYSICAL_PARAMETER:sensor = "INSTRUMENT_XXX").

File: sp041_20191205T1757-instrument-scalar_v2.cdl (hunk 4 above)
Comment (vturpin) [reply sp041_20191205T1757-instrument-scalar_v2.cdl]: @kerfoot I have updated the document accordingly

File: sp041_20191205T1757-instrument-scalar_v2.cdl (hunk 3 above)
Comment (vturpin) [reply sp041_20191205T1757-instrument-scalar_v2.cdl]: document updated accordingly

File: sp041_20191205T1757-instrument-scalar_v2.cdl (hunk 1 above)
Comment (vturpin) [reply sp041_20191205T1757-instrument-scalar_v2.cdl]: Document updated accordingly

File: sp041_20191205T1757-instrument-scalar_v2.cdl [hunk 5]
```diff
@@ -0,0 +1,417 @@
...
+        PSAL:vocabulary = "http://vocab.nerc.ac.uk/collection/OG1/current/PSAL/";
+		PSAL:standard_name = "sea_water_practical_salinity" ;
+		PSAL:units = "1" ;
+		PSAL:ancillary_variables = "PSAL_QC" ;
+		PSAL:valid_max = 40. ;
+		PSAL:valid_min = 0. ;
+		PSAL:coverage_content_type = "physicalMeasurement" ;
+		PSAL:coordinates = "time lon lat depth" ;
+		PSAL:comment = "Practical salinity computed using UNESCO 1983 algorithm" ;
+        PSAL:sensor = "/CTD";
```
[the remark this answers is not in the document]
Comment (vturpin) [reply sp041_20191205T1757-instrument-scalar_v2.cdl]: document updated accordingly

**Review by kerfoot — APPROVED** [review]
Changes look good. Thanks.

**Comment by vturpin** [comment]
@justinbuck  @castelao @glidermann @callumrollo could you approve this PR to make the file visible in the main repository. This will allow us to start finalizing the proto file.

**Review by glidermann — APPROVED** [review]

**Comment by glidermann** [comment]
looks good!

**Review by justinbuck — COMMENTED** [review]
Thank you for this, we can use it as the basis for a local test, aiming to to this in late November

**Review by callumrollo — COMMENTED** [review]
Looks great! Thanks Victor

File: sp041_20191205T1757-instrument-scalar_v2.cdl [hunk 6]
```diff
@@ -0,0 +1,417 @@
...
+		LONGITUDE:units = "degrees_east" ;
+		LONGITUDE:valid_max = "180" ;
+		LONGITUDE:valid_min = "-180" ;
+                LONGITUDE:interpolation_methodology = "TBD" ;
+                LONGITUDE:interpolation_methodology_vocabulary = "TBD" ;
+                LONGITUDE:interpolation_methodology_doi = "TBD" ;
+	float PRES(N_MEASUREMENTS) ;
+		PRES:_FillValue = -9999.9 ;
+		PRES:long_name = "Pressure" ;
+		PRES:vocabulary = "http://vocab.nerc.ac.uk/collection/OG1/current/PRES/";
```
Comment (emmerbodc) [inline sp041_20191205T1757-instrument-scalar_v2.cdl:81]: Just a minor correction the address should be - https

File: sp041_20191205T1757-instrument-scalar_v2.cdl [hunk 7]
```diff
@@ -0,0 +1,417 @@
...
+		PRES:ancillary_variables = "PRES_QC" ;
+		PRES:comment = "Sea water pressure, equals 0 at sea-level" ;
+                PRES:sensor = "INSTRUMENT_CTD";
+	byte PRES_QC(N_MEASUREMENTS) ;
+		PRES_QC:_FillValue = 0b ;
+		PRES_QC:long_name = "quality flag" ;
+	float DEPTH(N_MEASUREMENTS) ;
+		DEPTH:_FillValue = NaN ;
+		DEPTH:long_name = "Depth" ;
+		DEPTH:vocabulary = "http://vocab.nerc.ac.uk/collection/OG1/current/DEPTH/";
```
Comment (emmerbodc) [inline sp041_20191205T1757-instrument-scalar_v2.cdl:93]: https

File: sp041_20191205T1757-instrument-scalar_v2.cdl [hunk 8]
```diff
@@ -0,0 +1,417 @@
...
+		:time_coverage_end = "2019-12-05T18:05Z" ;
+		:program = "GOMO, IOOS" ;
+		:network = "California Underwater Glider Network" ;
+		:project = "California Underwater Glider Network" ;
+		:contributor_name = "Daniel Rudnick,Guilherme Castelao" ;
+		:contributor_role = "Principal Investigator, Data Curator" ;
+		:product_version = "v3" ;
+		:instrument = "Sea-Bird 41CP" ;
+		:metadata_link = "http://spraydata.ucsd.edu" ;
+		:ctd_make_model = "Sea-Bird 41CP" ;
```
Comment (emmerbodc) [inline sp041_20191205T1757-instrument-scalar_v2.cdl:287]: do we need to include this in the global attributes too?

File: sp041_20191205T1757-instrument-scalar_v2.cdl [hunk 9]
```diff
@@ -0,0 +1,417 @@
...
+		PLATFORM_MAKER:platform_maker_vocabulary = "TBD" ;
+	string PLATFORM_SERIAL_NUMBER ;
+		PLATFORM_SERIAL_NUMBER:long_name = "glider serial number" ;
+	string PLATFORM_CODE ;
+		PLATFORM_CODE:long_name = "nickname of the glider" ;
+	int PLATFORM_DEPTH_RATING ;
+		PLATFORM_DEPTH_RATING:long_name = "Depth limit in meters of the glider for this mission" ;
+	string WMO_IDENTIFIER ;
+		WMO_IDENTIFIER:long_name = "wmo id" ;
+        int INSTRUMENT_CTD ;
```
Comment (emmerbodc) [inline sp041_20191205T1757-instrument-scalar_v2.cdl:204]: Is this effectively an empty variable or are you expecting an int identifier for the CTD?

**Pull request merged**